Firestore/core/src/firebase/firestore/local/local_serializer.h

```diff
@@ -52,91 +53,56 @@ class LocalSerializer {
    * @brief Encodes a MaybeDocument model to the equivalent bytes for local
    * storage.
    *
+   * Any errors that occur during encoding are fatal.
+   *
+   * @param writer The serialized output will be written to the provided writer.
    * @param maybe_doc the model to convert.
-   * @param[out] out_bytes A buffer to place the output. The bytes will be
-   * appended to this vector.
-   * @return A Status, which if not ok(), indicates what went wrong. Note that
-   * errors during encoding generally indicate a serious/fatal error.
    */
-  // TODO(rsgowman): If we never support any output except to a vector, it may
-  // make sense to have LocalSerializer own the vector and provide an accessor
-  // rather than asking the user to create it first.
-  util::Status EncodeMaybeDocument(const model::MaybeDocument& maybe_doc,
-                                   std::vector<uint8_t>* out_bytes) const;
+  void EncodeMaybeDocument(nanopb::Writer* writer,
+                           const model::MaybeDocument& maybe_doc) const;
 
   /**
    * @brief Decodes bytes representing a MaybeDocument proto to the equivalent
    * model.
    *
-   * @param bytes The bytes to convert. It's assumed that exactly all of the
-   * bytes will be used by this conversion.
-   * @return The model equivalent of the bytes or a Status indicating what went
-   * wrong.
-   */
-  util::StatusOr<std::unique_ptr<model::MaybeDocument>> DecodeMaybeDocument(
-      const uint8_t* bytes, size_t length) const;
-
-  /**
-   * @brief Decodes bytes representing a MaybeDocument proto to the equivalent
-   * model.
+   * Check writer->status() to determine if an error occured while decoding.
```

Comment:
Hmm, this function takes a reader, not writer?

Reply:
Done.